File: fourier/series/chapter1/codes/1.1.py
import numpy as np
from scipy import signal
import matplotlib.pyplot as plt

#Wave Amplitude
A=5
#Wave Period
T = 0.02;
#Time Samples
t = np.linspace(-2.5*T,2.5*T,1e4)
#Plot wave
plt.plot(t, A/2*(1+signal.square(2*np.pi*t/T)))
plt.ylim(-1, 6)
plt.grid()
plt.xlabel('$t$')
plt.ylabel('$g(t)$')
#Save figure
plt.savefig('../figs/1.1.eps')
plt.show()

File: fourier/series/chapter1/codes/1.1.m
clear;
close;
clc;


%Square wave 
A = 5;
T = 0.02;


t = linspace(-2.5*T,2.5*T,1e4);
s = A/2*(1+square(2*pi*t/T));
plot(t,s,"Linewidth",4)
grid
axis([-0.06,0.06,-1,6])
xlabel('t')
ylabel('g(t)')

print -deps -color ../figs/1.1.eps

